core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,10 +60,12 @@
    * @param file the file the block belongs to
    * @param opType the OutStream's write type
    * @param blockIndex the index of the block in the file
+   * @param tachyonConf the TachyonConf instance for this file output stream.
    * @throws IOException
    */
-  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex) throws IOException {
-    super(file, opType);
+  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex, TachyonConf tachyonConf)
```

Comment:
Same here =) 

I think we could hang a lot of stuff in there and not just the configuration instance.